rpmio/rpmkeyring.cc

```diff
@@ -61,6 +69,55 @@ rpmKeyring rpmKeyringFree(rpmKeyring keyring)
     return NULL;
 }
 
+rpmKeyringIterator rpmKeyringInitIterator(rpmKeyring keyring, int unused)
+{
+    if (!keyring || unused != 0)
+	return NULL;
+
+    keyring = rpmKeyringLink(keyring);
+
+    rpmKeyringIterator iter = new rpmKeyringIterator_s {};
+    iter->keyringlock = new rdlock(keyring->mutex);
+    iter->iterator = keyring->keys.cbegin();
+    iter->keyring = keyring;
+    iter->current = NULL;
```

Comment:
Place all this initialization into the new {} directly, and then you don't need to (and shouldn't) manually allocate the lock, locking then happens automatically as a part of the initialization and then you don't need to (and shouldn't) free it manually either.

Reply:
Yeah, I am only slowly adjusting to this C++ thing. The whole file also calling new and then assigning stuff didn't quite push me in the right direction on this...

Reply:
Yup, the codebase isn't an example of how to do stuff in C++. 

I wouldn't have asked for it otherwise at all, but in this particular case it actually makes a difference: doing it like this allows you to take the lock right on construction.